crates/chia-sdk-driver/src/merkle_tree.rs

```diff
@@ -0,0 +1,179 @@
+use std::collections::HashMap;
+
+use chia_protocol::Bytes32;
+use sha2::{Digest, Sha256};
+
+const HASH_TREE_PREFIX: &[u8] = &[2];
+const HASH_LEAF_PREFIX: &[u8] = &[1];
+
+#[derive(Debug, Clone)]
+pub struct MerkleTree {
+    pub root: Bytes32,
+    pub proofs: HashMap<Bytes32, (u32, Vec<Bytes32>)>,
+}
+
+use std::fmt::Debug;
+
+#[derive(Debug, Clone)]
+pub enum BinaryTree<T> {
+    Leaf(T),
+    Node(Box<BinaryTree<T>>, Box<BinaryTree<T>>),
+}
+
+impl MerkleTree {
+    pub fn new(leaves: &[Bytes32]) -> Self {
+        if leaves.is_empty() {
+            return Self {
+                root: Bytes32::default(),
+                proofs: HashMap::new(),
+            };
+        }
+
+        let (root, proofs) = MerkleTree::build_merkle_tree(leaves);
+        Self { root, proofs }
+    }
+
+    fn build_merkle_tree(leaves: &[Bytes32]) -> (Bytes32, HashMap<Bytes32, (u32, Vec<Bytes32>)>) {
+        let binary_tree = MerkleTree::list_to_binary_tree(leaves).unwrap();
+        MerkleTree::build_merkle_tree_from_binary_tree(&binary_tree)
+    }
+
+    fn sha256(args: &[&[u8]]) -> Bytes32 {
+        let mut hasher = Sha256::new();
+        args.iter().for_each(|arg| hasher.update(arg));
+
+        let result = hasher.finalize();
+        let result: [u8; 32] = result.into();
+        Bytes32::from(result)
```

Comment:
`Bytes32::new(hasher.finalize().into())`
You also don't need the `into()` if you use the `Sha256` from `clvmr`